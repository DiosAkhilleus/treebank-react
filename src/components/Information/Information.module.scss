.information {
  background-color: rgb(242, 242, 242);

  padding-top: 0.2rem;
  padding-bottom: 0.2rem;
  padding-left: 0.4rem;
  padding-right: 0.4rem;
}
