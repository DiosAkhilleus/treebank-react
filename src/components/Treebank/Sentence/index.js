import { SentenceConsumer } from './Sentence';

export default SentenceConsumer;
