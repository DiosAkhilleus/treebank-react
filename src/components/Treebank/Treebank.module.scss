.treebank {
  background-color: rgb(242, 242, 242);

  path {
    stroke: #333;
    stroke-width : 0.5px;
    fill: none;
  }

  text {
    fill: #222;
    font-family: sans-serif;
  }

  :global {
    .nodes {
      fill: none;
    }

    .edgeLabels {
      text {
        font-size: 0.7rem;
      }
    }

    .nodes {
      text {
        font-size: 1rem;
      }
    }
  }
}
