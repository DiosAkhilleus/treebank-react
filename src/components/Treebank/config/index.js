import {
  getConfig,
  getColor,
} from './config';

export {
  getConfig,
  getColor,
};
