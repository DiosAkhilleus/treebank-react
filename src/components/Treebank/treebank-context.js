import { createContext } from 'react';

const TreebankContext = createContext();

export default TreebankContext;
