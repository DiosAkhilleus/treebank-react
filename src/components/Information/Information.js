import React, { Fragment } from 'react';
import { object } from 'prop-types';

import { wordType } from '../../types';

import styles from './Information.module.scss';

import { deconstructPostag } from '../Treebank/config';

const renderLemma = (lemma) => (
  <>
    <dt>Lemma</dt>
    <dd>{lemma}</dd>
  </>
);

const renderPostag = ([name, value]) => (
  <Fragment key={name}>
    <dt>{name}</dt>
    <dd>{value}</dd>
  </Fragment>
);

const Information = ({ active, config }) => {
  if (active) {
    const { $: { postag, lemma } } = active;

    return (
      <div className={styles.information}>
        <dl>
          {renderLemma(lemma)}
          {deconstructPostag(config, postag).map(renderPostag)}
        </dl>
      </div>
    );
  }

  return (
    <div />
  );
};

Information.propTypes = {
  active: wordType,
  // eslint-disable-next-line react/forbid-prop-types
  config: object.isRequired,
};

Information.defaultProps = {
  active: null,
};

export default Information;
