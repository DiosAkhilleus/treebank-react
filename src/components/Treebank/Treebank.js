import React, { useState } from 'react';
import { node, string } from 'prop-types';

import styles from './Treebank.module.scss';

import { xmlToJson } from '../../lib/parsing';
import { getConfig } from './config';

import TreebankContext from './treebank-context';

const sentenceFromJson = (treebankJson, chunk) => (
  treebankJson.treebank.sentence.find(({ $ }) => $.id && $.id === chunk)
);

const configFromJson = (treebankJson) => (
  getConfig(
    treebankJson.treebank.$['xml:lang'],
    treebankJson.treebank.$.format,
  )
);

const Treebank = ({ treebank, chunk, children }) => {
  const [active, setActive] = useState(null);
  const treebankJson = xmlToJson(treebank);
  const sentence = sentenceFromJson(treebankJson, chunk);
  const config = configFromJson(treebankJson);

  return (
    <TreebankContext.Provider value={{
      sentence,
      chunk,
      config,
      active,
      setActive,
    }}
    >
      <div className={styles.container}>
        {children}
      </div>
    </TreebankContext.Provider>
  );
};

Treebank.propTypes = {
  treebank: string.isRequired,
  chunk: string.isRequired,
  children: node,
};

Treebank.defaultProps = {
  children: null,
};

export default Treebank;
