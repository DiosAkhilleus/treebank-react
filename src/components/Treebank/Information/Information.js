import React from 'react';

import TreebankContext from '../treebank-context';

import Information from '../../Information';

const InformationWithContext = () => (
  <TreebankContext.Consumer>
    {({ active, config }) => <Information active={active} config={config} />}
  </TreebankContext.Consumer>
);

export default InformationWithContext;
