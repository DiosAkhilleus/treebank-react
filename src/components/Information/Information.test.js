import React from 'react';
import renderer from 'react-test-renderer';

import { getConfig } from '../Treebank/config';

import Information from './Information';

it('renders a sentence', () => {
  const active = {
    $: {
      id: '1',
      form: 'δικαστὰς',
      lemma: 'δικαστής',
      postag: 'n-p---ma-',
      relation: 'PNOM',
      head: '1',
    },
  };

  const component = (
    <Information
      active={active}
      config={getConfig('aldt', 'grc')}
    />
  );
  const tree = renderer.create(component).toJSON();

  expect(tree).toMatchSnapshot();
});
