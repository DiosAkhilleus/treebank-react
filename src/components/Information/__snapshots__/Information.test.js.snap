// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a sentence 1`] = `
<div
  className="information"
>
  <dl>
    <dt>
      Lemma
    </dt>
    <dd>
      δικαστής
    </dd>
    <dt>
      Part of Speech
    </dt>
    <dd>
      noun
    </dd>
    <dt>
      Number
    </dt>
    <dd>
      plural
    </dd>
    <dt>
      gender
    </dt>
    <dd>
      masculine
    </dd>
    <dt>
      Casus
    </dt>
    <dd>
      accusative
    </dd>
  </dl>
</div>
`;
